pkg/addonfactory/addonfactory.go

```diff
@@ -86,6 +95,12 @@ func (f *AgentAddonFactory) WithAgentHealthProber(prober *agent.HealthProber) *A
 	return f
 }
 
+// WithAgentHostedModeEnabledOption defines whether enable the agent hosted deploying mode.
+func (f *AgentAddonFactory) WithAgentHostedModeEnabledOption(option bool) *AgentAddonFactory {
```

Comment:
to make it simple, I would like to :
func (f *AgentAddonFactory) WithAgentHostedModeEnabledOption() *AgentAddonFactory {
	f.agentAddonOptions.HostedModeEnabled = true
	return f
}